pkg/provisioner/templates/kubernetes.go

```diff
@@ -147,12 +153,116 @@ echo "you can now access the cluster with:"
 echo "ssh -i <your-private-key> ubuntu@${INSTANCE_ENDPOINT_HOST}"
 `
 
+const NVKindTemplate = `
+
+: ${KIND_K8S_TAG:={{.Version}}}
+
+# The kind image to use. This image will be built if it is not available.
+: ${KIND_IMAGE:="kindest/node:${KIND_K8S_TAG}"}
+
+# The kubernetes tag to build the kind cluster from
+# From https://github.com/kubernetes/kubernetes/tags
+# See also https://hub.docker.com/r/kindest/node/tags
+: ${KIND_K8S_REPO:="https://github.com/kubernetes/kubernetes.git"}
+
+ARCH=$(uname -m)
+if [ "$ARCH" == "x86_64" ]; then
+  ARCH="amd64"
+fi
+if [ "$ARCH" == "aarch64" ]; then
+  ARCH="arm64"
+fi
+
+# Go
+# Set GO version
+GO_VERSION="1.23.6"
+wget https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz -O /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+sudo rm -rf /usr/local/go
+sudo tar -C /usr/local -xzf /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+
+# Add Go to PATH
+if ! grep -q 'export PATH="/usr/local/go/bin:$PATH"' ~/.bashrc; then
+  echo 'export PATH="/usr/local/go/bin:/$HOME/go/bin:$PATH"' >> ~/.bashrc
+fi
+export PATH="/usr/local/go/bin:$HOME/go/bin:$PATH"
+
+# Make
+install_packages_with_retry build-essential
+
+# Install kubectl
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+chmod +x kubectl
+sudo mv kubectl /usr/local/bin/
+
+# Install kind
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.26.0/kind-linux-${ARCH}
+chmod +x ./kind
+sudo install ./kind /usr/local/bin/kind
+
+# Install docker buildx, needed for building the kind image
+install_packages_with_retry docker-buildx-plugin
+
+# If an image ID already exists for the image we plan to build, we are done.
+EXISTING_IMAGE_ID="$(docker images --filter "reference=${KIND_IMAGE}" -q)"
+if [ "${EXISTING_IMAGE_ID}" != "" ]; then
+	exit 0
+fi
+
+# Create a temorary directory to hold all the artifacts we need for building the image
+TMP_DIR="$(mktemp -d)"
+cleanup() {
+    rm -rf "${TMP_DIR}"
+}
+trap cleanup EXIT
+
+# Set some build variables
+KIND_K8S_DIR="${TMP_DIR}/kubernetes-${KIND_K8S_TAG}"
+
+# Checkout the version of kubernetes we want to build our kind image from
+git clone --depth 1 --branch ${KIND_K8S_TAG} ${KIND_K8S_REPO} ${KIND_K8S_DIR}
+
+# Build the kind node image.
+# The default base image will be used unless it is specified using the
+# KIND_BASE_IMAGE environment variable.
+# This could be needed if new container runtime features are required.
+# Examples are:
+#   gcr.io/k8s-staging-kind/base:v20240213-749005b2
+#   docker.io/kindest/base:v20240202-8f1494ea
+kind build node-image \
+    --image "${KIND_IMAGE}" \
+    "${KIND_K8S_DIR}"
```

Comment:
Is it strictly required to build custom node images as part of this CI?

Reply:
no, removed